src/server/getters/get-market-price-history.ts

```diff
@@ -1,31 +1,30 @@
 import * as Knex from "knex";
+import * as _ from "lodash";
 import { Address, TimestampedPrice, MarketPriceHistory } from "../../types";
 import { sortDirection } from "../../utils/sort-direction";
 
-interface MarketPriceHistoryTradesRow {
+interface MarketPriceHistoryTradesRow extends TimestampedPrice {
   outcome: number;
-  price: string|number;
-  tradeTime: number;
 }
 
 // Input: MarketID
 // Output: { outcome: [{ price, timestamp }] }
 export function getMarketPriceHistory(db: Knex, marketID: Address|null|undefined, sortBy: string|null|undefined, isSortDescending: boolean|null|undefined, limit: number|null|undefined, offset: number|null|undefined, callback: (err?: Error|null, result?: MarketPriceHistory) => void): void {
-  const columnsToSelect: Array<string> = ["outcome", "price", "tradeTime"];
+  const columnsToSelect: Array<string> = ["outcome", "price", "tradeTime as timestamp"];
   let query: Knex.QueryBuilder = db.select(columnsToSelect).from("trades").where({ marketID }).orderBy(sortBy || "tradeTime", sortDirection(isSortDescending, "desc"));
   if (limit != null) query = query.limit(limit);
   if (offset != null) query = query.offset(offset);
   query.asCallback((err?: Error|null, tradesRows?: Array<MarketPriceHistoryTradesRow>): void => {
     if (err) return callback(err);
     if (!tradesRows || !tradesRows.length) return callback(null);
-    const marketPriceHistory: MarketPriceHistory = {};
-    tradesRows.forEach((trade: MarketPriceHistoryTradesRow): void => {
-      if (!marketPriceHistory[trade.outcome]) marketPriceHistory[trade.outcome] = [];
-      marketPriceHistory[trade.outcome].push({
-        price: trade.price,
-        timestamp: trade.tradeTime
-      });
-    });
+
+    // Group by outcome, drop "outcome" from result
+    const marketPriceHistory: MarketPriceHistory = _.mapValues(
```

Comment:
This is more declarative, but IMO less obvious / a bit harder to read.  Do you have a strong preference for this style?  If not, my vote is to stick with the old, dumb code -- remember that I have a great fondness for dumb code, the dumber the better! ;)